modules/tensor_mechanics/src/materials/ComputeReturnMappingStress.C

```diff
@@ -77,10 +92,18 @@ ComputeReturnMappingStress::computeQpStress()
   RankTwoTensor strain_increment = _strain_increment[_qp];
   RankTwoTensor stress_new;
   updateQpStress(strain_increment, stress_new);
-  _elastic_strain[_qp] = _rotation_increment[_qp] * (strain_increment + _elastic_strain_old[_qp]) *
-                         _rotation_increment[_qp].transpose();
-  _stress[_qp] = _rotation_increment[_qp] * stress_new * _rotation_increment[_qp].transpose();
 
+  if (_perform_finite_strain_rotations)
+  {
+    _elastic_strain[_qp] = _rotation_increment[_qp] *
```

Comment:
After looking at this again, I think you have the right tensor rotations here.  ComputeFiniteStrain rotates `total_strain` and `mechanical_strain` (which is assigned to `elastic_strain` in ComputeFiniteStrainElasticStress, so you don't have to rotate `elastic_stress` in that model, but you do need to do it here.)